### ISSUE TITLE
2.2.0 - Imcompatibility with ReShade DX11

### ISSUE THREAD

**Issue by WyvrnRipsnarl** [Friendly0Fire/GW2Radial#284]
Cannot boot Radial with ReShade (and vice-versa), under DX11.
NOTE: 
- Similar State as when GW2 DX11 BETA has came out.
- Stable in DX9 version.
- Both ReShade and Radial works Individually, in DX11 though.

Attempted Manual Installation and Automatic Addon Modloader and absolutely nothing matched.

**Comment by Friendly0Fire** [comment]
How are you setting up Reshade?

**Comment by WyvrnRipsnarl** [comment]
Via the Installer and Selecting DX10/11/12
Once installed, either Radial OR ReShade will work.

I have try to install Radial both Manually and via the Addon Modloader.
So I don't know where the imcompatibility comes from, other than the Result.

**Comment by Friendly0Fire** [comment]
If the problem arises when using SelectRenderer with Reshade correctly installed, then I don't know what's wrong. This may not be a GW2Radial issue specifically, but either way it's something that I'm not sure I'll be able to debug in the near future so you may want to turn off Reshade for the time being.

**Comment by tildebyte** [comment]
GShade + Radial (installed manually) works without issue here.

**Comment by ludwigma** [comment]
Gshade + radial works for me too !

**Comment by InsaniusMax** [comment]
> Gshade + radial works for me too ! installed via addon manager arcdps / radial / selectrenderer , install gshade in the addons/reshade folder ... in select renderer select reshade dx11

How did you install Gshade in the addons/reshade folder? Since it asks me for the executable upon installing, it always installs to the GW2 root folder.

**Comment by ludwigma** [comment]
just create a dummy text file with the name gw2.exe or whatever you want ...it just searches for a exe ,  just fake it :-)

**Comment by InsaniusMax** [comment]
> just create a dummy text file with the name gw2.exe or whatever you want ...it just searches for a exe , just fake it :-)

Might I ask what settings you selected on installation? At the moment, I have these selected and none are working:
![image](https://user-images.githubusercontent.com/100453677/158815343-4b07e727-4aee-4554-b2ec-d47f1e8031a6.png)

Also, do you first install reshade and then install gshade to the same folder?

**Comment by ludwigma** [comment]
yes these settings are correct, you point at the folder /addons/reshade  , but your are not installing reshade only gshade.
and make sure there are symlinks in place...should be a dxgi.dll link to the gshade64.dll in your programs folder !

**Comment by InsaniusMax** [comment]
> yes these settings are correct, you point at the folder /addons/reshade , but your are not installing reshade only gshade. and make sure there are symlinks in place...should be a dxgi.dll link to the gshade64.dll in your programs folder !

Thanks, this mostly fixed it for me, but for some reason, the preset settings are not showing up in the in-game GShade editor, so I can't actually change anything atm. Any idea what might cause this?

**Comment by ludwigma** [comment]
yes this is a reshade problem. before you load in example the tekkit preset. load up gshade menu go to settings and select the shaders and textures folders via hand , then everything appears if you press reload at the bottom. selecting the correct folder is easy because there is something like a browser integrated. have fun !

**Comment by InsaniusMax** [comment]
That did it for me, thanks :) Time to turn off in-game post-processing for that sweet FPS boost.

**Comment by TommyTx3** [comment]
> That did it for me, thanks :) Time to turn off in-game post-processing for that sweet FPS boost.

Hey, may I reach out to you somewhere else for a little tutorial on how you fixed this? I've tried everything besides this but I'm not good enough to understand the procedure :)
Either ingame or Discord is fine by me.

**Comment by InsaniusMax** [comment]
> > That did it for me, thanks :) Time to turn off in-game post-processing for that sweet FPS boost.
> 
> Hey, may I reach out to you somewhere else for a little tutorial on how you fixed this? I've tried everything besides this but I'm not good enough to understand the procedure :) Either ingame or Discord is fine by me.

Sure, add me on Discord. Ian#1023

**Comment by WyvrnRipsnarl** [comment]
Attempted the above methods, but still nothing conclusive except failure to make them both work together... Can someone make a detail guide to install this, because this is getting frustrating...

**Comment by TommyTx3** [comment]
> Attempted the above methods, but still nothing conclusive except failure to make them both work together... Can someone make a detail guide to install this, because this is getting frustrating...

I can help you! Add me on Discord I'm TommyTx3#6538 :)

**Comment by Eggy699** [comment]
Could anyone help me with installing both. Having issues dxgi.dll file. When installing using the mod manger it replaced the file so the GShader will no longer work (vis versa). Do you have to somehow merge the files together?

**Comment by Friendly0Fire** [comment]
Is this still an issue or has this been resolved?

**Comment by WyvrnRipsnarl** [comment]
Issue is still occuring.

**Comment by Friendly0Fire** [comment]
Did you work with either of the people who offered help?

**Comment by WyvrnRipsnarl** [comment]
I did.

**Comment by Friendly0Fire** [comment]
... And? What further steps did you try?

**Comment by WyvrnRipsnarl** [comment]
We tried the full reinstalls, with and without modloader. Fresh GW2 Installs, both DX9 and DX11. And it is still one or the other but not both at the same time.

**Comment by ExreaLz** [comment]
I think I might found the issue, both reshade and gw2radial for dx11 are using dxgi.dll, so when gw2radial work, reshade not, because dxgi.dll is now overwritten for gw2radial ? how to make them both using the same dxgi.dll or maybe make gw2radial dx11 use other renamed dxgi.dll ?

Here how I test it, install gw2radial dx11 through addon manager with working reshade installed, gw2radial works but reshade not, then close the game and reinstall reshade, it pop up a error said dxgi.dll is not for reshade, so remove it and reinstall reshade, now reshade work but gw2radial not working again, hope this explanation help to figure out how to solve it ? :)

**Comment by knoxfighter** [comment]
If you want to use reshade/gshade with gw2radial, you have to have a proper installation of the addon manager. When done correctly, only the addon loader is creating the dxgi.dll.
If you have a clean GW2 installation:

### Installing gShade/ReShade with the addon-manager (first setup):
1. install gShade, so the `dxgi.dll` is created.
2. move the `dxgi.dll` out of the directory (we need it later on and it is in the way atm)
3. install arcdps with the addon-manager
4. install `lib_imgui` and `SelectRender` in the addon-manager
5. create a directory called `reshade` inside the folder `addons`
6. move the `dxgi.dll` you moved away into the folder `reshade`
7. start the game (a popup of SelectRender will be opened)
8. Click on `Change` in that window
9. Select `ReShade` under the `DX11` category
10. Click `Apply`
11. restart the game

**Comment by ExreaLz** [comment]
> If you want to use reshade/gshade with gw2radial, you have to have a proper installation of the addon manager. When done correctly, only the addon loader is creating the dxgi.dll. If you have a clean GW2 installation:
> ### Installing gShade/ReShade with the addon-manager (first setup):
> 
>     1. install gShade, so the `dxgi.dll` is created.
> 
>     2. move the `dxgi.dll` out of the directory (we need it later on and it is in the way atm)
> 
>     3. install arcdps with the addon-manager
> 
>     4. install `lib_imgui` and `SelectRender` in the addon-manager
> 
>     5. create a directory called `reshade` inside the folder `addons`
> 
>     6. move the `dxgi.dll` you moved away into the folder `reshade`
> 
>     7. start the game (a popup of SelectRender will be opened)
> 
>     8. Click on `Change` in that window
> 
>     9. Select `ReShade` under the `DX11` category
> 
>     10. Click `Apply`
> 
>     11. restart the game
> 
> 
> Now you can install radial in the addon manager as well and it should work out of the box.

Thanks for the guide, really appreciated it, seems working but there's a issue.

Installing gw2radial dx11 with the addon-manager + Reshade:

    1) Install Reshade
    2) Move dxgi.dll in guild wars 2 game folder to somewhere else, I moved mine to desktop.
    3) Install lib_imgui and SelectRender in the addon-manager.
    4) Create a folder called reshade inside guild wars 2 game folder named addons.
    5) Now, move the dxgi.dll you moved away earlier into the folder in addons named reshade.
    6) Start the game and SelectRender menu will pop up.
    7) Click Change in that window to select ReShade under the DX11 category and click Apply
    9) Exit the game
  10) Now, reopen addon-manager to install gw2radial dx11.

Note: have not tested playing the game in long period yet, seems like working but every time I exit the game, there's error pop up for gw2-64.exe asking me to send crash report to Arenanet.

**Comment by Vetkiin** [comment]
Just wanted to comment that I'm having the **exact same issue** as [ExreaLz](https://github.com/ExreaLz).

Running Reshade 5.4.2 with custom shaders like RTGI and unable to get GW2Radial to fully cooperate since I'm missing some shaders that return when uninstalling GW2Radial.

Unfortunately that stops me from using GW2Radial for the moment until a hotfix is released since I value reshade more than GW2Radial but love both.

Hope this helps and wish you a great weekend!

**Comment by dionatan-g** [comment]
Same problem, waiting for a fix.

**Comment by Friendly0Fire** [comment]
This issue is marked as off-topic. It will not be fixed because I have nothing to fix, but remains open for people to look for solutions.

**Comment by plopek** [comment]
To resolve issue with ReShade Version 5.9.1 and ArcDPS (20230823-025443) do the following.
1) Install Reshade.
2) Rename dxgi.dll to d3d11.dll in game folder and move it to /bin64/cef.
3) Install ArcDPS.
4) Rename now d3d11.dll to dxgi.dll
5) Start the game. Both Configuration overlays will pop up :)

**Comment by Friendly0Fire** [comment]
Please don't do that, you're breaking the addon loader by doing this.

**Comment by knoxfighter** [comment]
Seems like the latest iteration of my install instructions can be posted here:

## Installing ReShade with the addon-manager:
1. open the addon manager
2. install `lib_imgui` and `SelectRender` in the addon-manager (and all additional addons you want)
3. delete the `dxgi.dll` next to the gw2 executable
4. install ReShade, select gw2 as game and install like normal
5. create a directory called `reshade` inside the folder `addons`
6. move the `dxgi.dll`, `ReShade.ini`, `ReShade.log` and the folder `reshade-shaders` from the gw2 folder into the created `reshade` folder
7. in the addon manager click on "Reinstall loader"
8. start the game (a popup of SelectRender will be opened)
9. Click on `Change` in that window
10. Select `ReShade` under the `DX11` category
11. Click `Apply`
12. restart the game

**Comment by plopek** [comment]
> Please don't do that, you're breaking the addon loader by doing this.

@Friendly0Fire Well, I had no idea there is more elegant solution for that. Since I didnt use addon-menager I didnt really know it could break that functionality.


@knoxfighter thank you for the workaround.
